Processing data in models

sambeck87 opened the pull request:
- Create a User method that returns the 3 most recent posts for a given user.
- Created a Post method that updates the posts counter for a user. 
- Create a Post method that returns the 5 most recent comments for a given post. 
- Create a Comment method that updates the comments counter for a post. 
- Create a Like method that updates the likes counter for a post.

sambeck87 committed:
Create an User method that returns the 3 most recent posts for a given user.
Created a Post method that updates the posts counter for a user.
Create a Post method which returns the 5 most recent comments for a given post.
Create a Comment method that updates the comments counter for a post.
Create a Like method that updates the likes counter for a post.

danmainah commented on README.md:
```diff
@@ -42,10 +42,11 @@
 
 This project allows you:
```
- [ ] [optionakl] I would suggest you change this from this project to this milestone since the project has a larger scope than what you have listed.

danmainah commented on README.md:
````diff
@@ -120,6 +121,7 @@ bundle install
 ```
+rails c
 
 ```
````
- [ ] @sambeck87 great work on your documentation however the license link is broken kindly fix it.

danmainah commented on app/models/user.rb:
```diff
@@ -1,3 +1,9 @@
 class User < ApplicationRecord
   has_many :posts
+  has_many :comments
+  has_many :likes
```
- [ ] @sambeck87 great work on your model, kindly add a foreign key that will be used to reference the user in this case 'author_id'

sambeck87 replied:
@danmainah, correct me if I'm wrong but I can't use a foreign key into User to reference the user_id. The foreign key "author" is in the Post class that references user_id

danmainah replied:
- [ ] @sambeck87 but you need also to specify the same in the user model, for not only post but also like and comment.

danmainah commented on app/models/like.rb:
```diff
@@ -1,4 +1,9 @@
 class Like < ApplicationRecord
   belongs_to :user
```
- [ ] @sambeck87 kindly add a class name as you did in the comment and post models

danmainah commented on app/models/post.rb:
```diff
@@ -1,3 +1,14 @@
+  def self.update_posts_counter(name)
+    user = User.find_by(name:)
+    user.update(posts_counter: Post.where(author_id: User.find_by(name:).id).length)
+  end
```
- [ ] [optional] @sambeck87 You could also use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - `after_save` to call the `#update_post_counter` method every time a new post has been saved to the database, then make the `update_posts_counter` method private , this could also apply to other counters.
```suggestion
  # app/models/post.rb
  after_save :update_post_counter

  private

  def update_post_counter
    author.increment!(:posts_counter)
  end

```

danmainah reviewed (changes requested):
Hi @sambeck87  , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
✔️ No linter errors
✔️ Gitflow was correctly used
✔️  Good design


# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

sambeck87 committed:
Fix the licence link
Add references in User model
Add a class_name User into Like model
Change the method that increment the number of likes, posts, and comments

sambeck87 committed:
Fix linters

Owusu-Desmond reviewed (approved):
Hi @sambeck87 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

<img src="https://media2.giphy.com/media/26BRBKqUiq586bRVm/giphy.gif"/>

## Highlights
📌 No Linters error ✅
📌 Descriptive pull request✅
📌 Good commit messages ✅
📌 Professional README file ✅

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
